### PR TITLE
th-164: Create a mask for the phone fields for the forms on the Sign up Business and Sign up Customer pages

### PR DESCRIPTION
**Pull request by albsbz**
Add phone field to input component, maybe scrolbar should be updated in future for the same from one in dashboard PR
Phone
phone3
phone2

### CODE REVIEW

**Review by DmytroVoronkov — APPROVED** [review]

**Review by denred — APPROVED** [review]

File: frontend/src/libs/components/input/input.tsx [hunk 1]
```diff
@@ -41,32 +45,73 @@ const Input = <T extends FieldValues>({
...
+      hasError && styles.error,
+    ],
+    [hasError, hasValue, isDisabled],
+  );
+
+  const defaultInputStyles = useMemo(
+    () => [styles.input, styles.inputPadding, ...controlledInputStyles],
+    [controlledInputStyles],
+  );
+
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/input/input.tsx:103]: do u need to wrap it in useMemo?

Comment (albsbz) [reply frontend/src/libs/components/input/input.tsx:103]: yes, eslint suggest it: The 'controlledInputStyles' array makes the dependencies of useCallback Hook (renderPhoneInput) change on every render. To fix this, wrap the initialization of 'controlledInputStyles' in its own useMemo()

**Review by h0wter — APPROVED** [review]

**Review by AlexVlaso — APPROVED** [review]

**Review by MaxHopaytsa — APPROVED** [review]

**Review by kalinkaaaa14 — DISMISSED** [review]

File: frontend/src/libs/components/input/input.tsx [hunk 2]
```diff
@@ -89,42 +86,98 @@ const Input = <T extends FieldValues>({
...
+        inputClass={getValidClassNames(
+          styles.input,
+          styles.phoneInput,
+          ...controlledInputStyles,
+        )}
+        buttonClass={styles.phoneButton}
+        dropdownClass={styles.phoneDropdown}
+        inputProps={{
+          id,
+        }}
```
Comment (Some14u) [inline frontend/src/libs/components/input/input.tsx]:
```suggestion
        inputProps={{ id }}
```

File: frontend/src/libs/components/input/input.tsx [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
   type FieldErrors,
   type FieldPath,
   type FieldValues,
   type UseFormSetError,
+  Controller,
 } from 'react-hook-form';
+import PhoneInput from 'react-phone-input-2';
 
 import { type InputType } from '~/libs/enums/enums.js';
 import { IconName } from '~/libs/enums/icon-name.enum.js';
```
Comment (Some14u) [inline frontend/src/libs/components/input/input.tsx]: import { type InputType, IconName } from '~/libs/enums/enums.js';
here and in other 10 places all over the project.

**Review by Some14u — CHANGES_REQUESTED** [review]

**Review by denred — APPROVED** [review]

**Review by h0wter — APPROVED** [review]

**Review by PavloSukhinin — APPROVED** [review]

**Pull request merged**